src/main/java/uk/gov/companieshouse/api/accounts/util/logging/ControllerLoggingAspect.java

```diff
@@ -0,0 +1,87 @@
+
+import static uk.gov.companieshouse.api.accounts.CompanyAccountsApplication.APPLICATION_NAME_SPACE;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.AfterThrowing;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Aspect
+@Component
+public class ControllerLoggingAspect {
+
+  private static final String REQUEST_ID = "X-Request-Id";
+
+  private static final String ERIC_IDENTITY = "Eric-Identity";
+
+  @Pointcut("execution(* uk.gov.companieshouse.api.accounts.controller.*.*(..))")
+  private void controllerMethods() {
+    //Defines the point cut for controller methods, the method body kept empty on purpose.
+  }
+
+  @Around("controllerMethods()")
+  public void logPerformanceStats(ProceedingJoinPoint joinPoint) throws Throwable {
+    long startTime = System.currentTimeMillis();
+    //Do not change, the target method will not progress without this.
+    Object result = joinPoint.proceed();
+    long responseTime = System.currentTimeMillis() - startTime;
+
+    LogBuilder logBuilder = getLogBuilder(joinPoint);
+    logBuilder.logStartOfRequestProcessing(String.format("Start Request %s",
+        methodName(joinPoint)));
+
+    logBuilder.logEndOfRequestProcessing(String.format("End of Request %s Successfully Completed",
+        methodName(joinPoint)), statusCode(result), responseTime);
+  }
+
+  private String methodName(JoinPoint joinPoint) {
+    return joinPoint.getSignature().getName();
+  }
+
+  private String requestId(HttpServletRequest request) {
+    return request.getHeader(REQUEST_ID);
+  }
+
+  private String userId(HttpServletRequest request) {
+    return request.getHeader(ERIC_IDENTITY);
+  }
+
+  private LogBuilder getLogBuilder(JoinPoint joinPoint) {
+    HttpServletRequest request = getRequest(joinPoint);
+    String accountId = "";
+    return new LogBuilderImpl(requestId(request), userId(request),
+        transactionId(joinPoint), accountId);
+  }
+
+  /**
+   * Helper to get the HttpServletRequest so that we can get the requestID and the userID.
+   */
+  private HttpServletRequest getRequest(JoinPoint joinPoint) {
+    return (HttpServletRequest) joinPoint.getArgs()[0];
+  }
+
+  private String transactionId(JoinPoint joinPoint) {
+    return joinPoint.getArgs()[1].toString();
```

Comment:
Is this forcing controller method signatures to be a certain format?
```java
public void [name](request, transactionId, ...)
```

The AOP shouldn't be controlling the method signatures. Will this work for other controllers? e.g.
```java
public ResponseEntity create(@Valid @RequestBody SmallFull smallFull, HttpServletRequest request)
```

Reply:
agreed, the request parameters can be identified based on their type, ex HttpServletRequest, SmallFull etc. However for the Ids (transactionId, accountId) should we log them as an array of Strings associated with a key=resources ?